src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -89,41 +92,63 @@ def load_arguments(self, _):
         c.argument('resource_group_name', arg_type=resource_group_name_type, help='the resource group where the policy will be applied')
 
     with self.argument_context('policy definition', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
-        c.argument('policy_definition_name', arg_type=existing_policy_definition_name_type)
         c.argument('rules', help='JSON formatted string or a path to a file with such content', type=file_type, completer=FilesCompleter())
-        c.argument('display_name', help='display name of policy definition')
-        c.argument('description', help='description of policy definition')
-        c.argument('params', help='JSON formatted string or a path to a file or uri with parameter definitions', type=file_type, completer=FilesCompleter(), min_api='2016-12-01')
+        c.argument('display_name', help='Display name of policy definition.')
+        c.argument('description', help='Description of policy definition.')
+        c.argument('params', help='JSON formatted string or a path to a file or uri with parameter definitions.', type=file_type, completer=FilesCompleter(), min_api='2016-12-01')
         c.argument('metadata', min_api='2017-06-01-preview', nargs='+', validator=validate_metadata, help='Metadata in space-separated key=value pairs.')
+        c.argument('subscription_id', arg_type=subscription_id_type)
+        c.argument('management_group', arg_type=management_group_name_type)
 
     with self.argument_context('policy definition create', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
         from azure.mgmt.resource.policy.models import PolicyMode
-        c.argument('name', options_list=('--name', '-n'), help='name of the new policy definition')
-        c.argument('mode', arg_type=get_enum_type(PolicyMode), options_list=('--mode', '-m'), help='mode of the new policy definition.', min_api='2016-12-01')
+        c.argument('name', options_list=('--name', '-n'), help='Name of the new policy definition.')
+        c.argument('mode', arg_type=get_enum_type(PolicyMode), options_list=('--mode', '-m'), help='Mode of the new policy definition.', min_api='2016-12-01')
+
+    with self.argument_context('policy definition show', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
+        c.argument('policy_definition_name', arg_type=existing_policy_definition_name_type)
+
+    with self.argument_context('policy definition update', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
+        c.argument('policy_definition_name', arg_type=existing_policy_definition_name_type)
+
+    with self.argument_context('policy definition delete', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
+        c.argument('policy_definition_name', arg_type=existing_policy_definition_name_type)
```

Comment:
These are all redundant and should just be under a single context `policy definition`.

Reply:
Done.